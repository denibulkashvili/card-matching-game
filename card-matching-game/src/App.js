import React, { Component } from 'react';
import Button from './components/ButtonReusable';
import DisplayGrid from './components/DisplayGrid';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gameState: gameState.Stopped,
    };
  }
  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Card Matching Game</h1> 
        <DisplayGrid />
        <Button />
      </div>
    );
  }
}

export default App;
