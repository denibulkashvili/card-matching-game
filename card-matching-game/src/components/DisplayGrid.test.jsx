import React from 'react';
import '../tests/setup';
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme';
import Display from './DisplayGrid';
import App from '../App';

describe('<Display /> component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Display />);
    expect(tree).toMatchSnapshot();
  });
  it('renders 12 images', () => {
    const appWrapper = mount(<App />)
    const button = appWrapper.find('.btn');
    button.simulate('click');
    const images = appWrapper.find('.image');
    expect(images).toHaveLength(12);
  });
  
})