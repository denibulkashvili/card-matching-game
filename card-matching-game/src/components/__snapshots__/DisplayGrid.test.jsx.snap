// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> component matches snapshot 1`] = `
<div
  id="display"
>
  <div
    id="image-grid"
  >
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="/img/cover.png"
    />
  </div>
</div>
`;
